### PR TITLE
Fix some more phpstan 1 errors

### PR DESCRIPTION
**Pull request by herndlm**

### CODE REVIEW

**Comment by herndlm** [comment]
wait sorry, I broke tests :/

File: src/Composer/Command/InitCommand.php [hunk 1]
```diff
@@ -487,15 +487,6 @@ public function parseAuthorString($author)
         );
     }
 
-    /**
-     * @param string $name
-     * @return list<array{name: string, description: ?string}>
-     */
-    protected function findPackages($name)
```
Comment (herndlm) [inline src/Composer/Command/InitCommand.php:494]: Is this internal enough to be removed? If not the return type of `RepositoryInterface::search` simply needs to be duplicated

Comment (Seldaek) [reply src/Composer/Command/InitCommand.php:494]: I guess time will tell but I'd say it's fine.

**Comment by Seldaek** [comment]
Thanks

**Comment by herndlm** [comment]
the remaining errors are quite tricky. Some might be even phpstan bugs or should be ignored potentially. Some might be easily fixable by you with Composer internal knowledge hopefully

**Comment by Seldaek** [comment]
OK no worries I'll take a look, it's only a few left thanks to your help here!

**Pull request merged**